### PR TITLE
fix(datastore): lazy to init table name for eval & replace name with id

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description

1. lazy to init table name for eval on the client
2. fix bug: replace name with id for eval summary table name 

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/wrapper.py [hunk 1]
```diff
@@ -85,19 +85,27 @@ def __init__(self, eval_id: str, project: str, instance: str = ""):
...
-        )
+    def _fetch_table_name(self, table: str) -> str:
+        with self._lock:
+            _table_name = self._tables.get(table)
+            if _table_name is None:
+                _table_name = data_store.gen_table_name(
+                    project=self.project,
+                    table=table,
+                    instance_uri=self.instance,
+                )
```
Comment (jialeicui) [inline client/starwhale/api/_impl/wrapper.py:106]: It seems like there will be a network request for each table name getting.
Add a cache decorator for `_get_remote_project_id` ?

File: client/starwhale/api/_impl/wrapper.py [hunk 2]
```diff
@@ -109,7 +117,10 @@ def log_result(
         record = {"id": data_id, "result": _serialize(result) if serialize else result}
         for k, v in kwargs.items():
             record[k.lower()] = _serialize(v) if serialize else v
-        self._log(self._results_table_name, record)
+
+        self._log(
```
Comment (tianweidut) [inline client/starwhale/api/_impl/wrapper.py:121]: `Evaluation` Class override `_log` and `_flush` method, add add `self._fetch_table_name` logical?

```
class Evalution(Logger):

def _log(self, table_name: str, record: Dict[str, Any]) -> None:
    table_name = self._fetch_tablename(table_name)
    super()._log(table_name, record)
```

File: client/starwhale/api/_impl/wrapper.py [hunk 3]
```diff
@@ -85,19 +85,27 @@ def __init__(self, eval_id: str, project: str, instance: str = ""):
...
-        self._init_writers([self._results_table_name, self._summary_table_name])
+        self._init_writers([])
 
-    def _get_datastore_table_name(self, name: str) -> str:
-        return data_store.gen_table_name(
-            project=self.project,
-            table=f"eval/{self.eval_id[:VERSION_PREFIX_CNT]}/{self.eval_id}/{name}",
-            instance_uri=self.instance,
-        )
+    def _fetch_table_name(self, table: str) -> str:
```
Comment (tianweidut) [inline client/starwhale/api/_impl/wrapper.py:98]: use another meaningful word to replace `_fetch_table_name`?

File: client/starwhale/api/_impl/wrapper.py (hunk 1 above)
Comment (goldenxinxing) [reply client/starwhale/api/_impl/wrapper.py:106]: done

File: client/starwhale/api/_impl/wrapper.py (hunk 2 above)
Comment (goldenxinxing) [reply client/starwhale/api/_impl/wrapper.py]: fixed

File: client/starwhale/api/_impl/wrapper.py (hunk 3 above)
Comment (goldenxinxing) [reply client/starwhale/api/_impl/wrapper.py]: fixed

**Review by tianweidut — APPROVED** [review]

**Pull request merged**